### ISSUE TITLE
Investigate error with multiple certificates in a key- or truststore

### ISSUE THREAD

**Issue by lfrancke** [stackabletech/issues#522]
After the [AutoTLS CA rotation PR](https://github.com/stackabletech/secret-operator/issues/93) has been merged we began seeing these errors in HBase:

```
Failed construction RegionServer
java.io.IOException: Problem starting http server
	at org.apache.hadoop.hbase.http.HttpServer.start(HttpServer.java:1140)
	at org.apache.hadoop.hbase.http.InfoServer.start(InfoServer.java:151)
	at org.apache.hadoop.hbase.regionserver.HRegionServer.putUpWebUI(HRegionServer.java:2235)
	at org.apache.hadoop.hbase.regionserver.HRegionServer.<init>(HRegionServer.java:694)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.apache.hadoop.hbase.regionserver.HRegionServer.constructRegionServer(HRegionServer.java:3203)
	at org.apache.hadoop.hbase.regionserver.HRegionServerCommandLine.start(HRegionServerCommandLine.java:63)
	at org.apache.hadoop.hbase.regionserver.HRegionServerCommandLine.run(HRegionServerCommandLine.java:87)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:82)
	at org.apache.hadoop.hbase.util.ServerCommandLine.doMain(ServerCommandLine.java:152)
	at org.apache.hadoop.hbase.regionserver.HRegionServer.main(HRegionServer.java:3221)
Caused by: java.lang.IllegalStateException: KeyStores with multiple certificates are not supported on the base class org.apache.hbase.thirdparty.org.eclipse.jetty.util.ssl.SslContextFactory. (Use org.apache.hbase.thirdparty.org.eclipse.jetty.util.ssl.SslContextFactory$Server or org.apache.hbase.thirdparty.org.eclipse.jetty.util.ssl.SslContextFactory$Client instead)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.ssl.SslContextFactory.newSniX509ExtendedKeyManager(SslContextFactory.java:1288)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.ssl.SslContextFactory.getKeyManagers(SslContextFactory.java:1270)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.ssl.SslContextFactory.load(SslContextFactory.java:372)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.ssl.SslContextFactory.doStart(SslContextFactory.java:243)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.server.SslConnectionFactory.doStart(SslConnectionFactory.java:97)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.server.AbstractConnector.doStart(AbstractConnector.java:321)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:81)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.server.Server.doStart(Server.java:401)
	at org.apache.hbase.thirdparty.org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.apache.hadoop.hbase.http.HttpServer.start(HttpServer.java:1109)
	... 13 more
```

As a developer I want to understand whether HBase is complaining about Key- or Truststores.
If it is about Keystores we need to understand why multiple certificates can be added to them.
If it is about Truststores we need to understand why that doesn't work as it should be fairly normal for a Truststore to have multiple CAs.

The scope of this issue is purely about the investigation part and a potential fix can be made in a separate issue.

**Comment by nightkr** [comment]
As far as I can tell, this doesn't actually have anything to do with rotation.

The error comes from https://github.com/jetty/jetty.project/blob/jetty-9.4.30.v20200611/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java#L1274-L1278, which is invoked from https://github.com/jetty/jetty.project/blob/jetty-9.4.30.v20200611/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java#L1249-L1258. One of the conditions that triggers this is `_certHosts.size() > 1`. `_certHosts` has [one entry per hostname](https://github.com/jetty/jetty.project/blob/jetty-9.4.30.v20200611/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java#L361-L364) (both [CN](https://github.com/jetty/jetty.project/blob/jetty-9.4.30.v20200611/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/X509.java#L99) and [SAN](https://github.com/jetty/jetty.project/blob/jetty-9.4.30.v20200611/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/X509.java#L84)), meaning that this condition should be triggered by _any_ certificate that has a SAN.

Since we have always used SAN as our certificate identity mechanism, this error _should_ always have been triggered.

Summarized from Slack: https://stackable-workspace.slack.com/archives/C033A863YPL/p1707301087529009

**Comment by lfrancke** [comment]
We have investigated this issue and found that it was unrelated to TLS CA Rotation.

Quick summary:
- A few weeks ago we merged Kerberos support for HBase
- This included enabling TLS for the HBase web UI
- Tests were running successfully against HBase 2.4.17, 2.4.12 was never tested
- A few weeks later doing something entirely different we ran the tests against 2.4.12 and that failed, because of it being so close to the merge of the CA rotation we jumped to the wrong conclusions
- The fix is to use either HBase 2.4.12 or (what we did) backport the fix to 2.4.12

FYI: We will remove 2.4.12 anyway for the next SDP release.
